src/averageAges.js

```diff
@@ -55,10 +65,49 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const peopleFiltered = people.filter(person => onlyWithSon
+    ? person.sex === 'm'
+      && person.mother
+    : person.mother
+      && people
+        .find(pers =>
```

Comment:
Good job with a solution. But maybe you should rid of one of the find methods because you have another similar method further in the code. And also your code working fine without this find🙃